skimage/feature/tests/test_peak.py

```diff
@@ -85,6 +85,56 @@ def test_num_peaks():
     assert (3, 5) in peaks_limited
 
 
+def test_num_peaks_and_labels():
+    image = np.zeros((7, 7), dtype=np.uint8)
+    labels = np.zeros((7, 7), dtype=np.uint8) + 20
+    image[1, 1] = 10
+    image[1, 3] = 11
+    image[1, 5] = 12
+    image[3, 5] = 8
+    image[5, 3] = 7
+    peaks_limited = peak.peak_local_max(
+        image, min_distance=1, threshold_abs=0, labels=labels)
+    assert len(peaks_limited) == 5
+    peaks_limited = peak.peak_local_max(
+        image, min_distance=1, threshold_abs=0, labels=labels, num_peaks=2)
+    assert len(peaks_limited) == 2
+
+def test_num_peaks_per_labels():
```

Comment:
I'm not sure about this test.
On the one hand, it doesn't check if `num_peaks` and `num_peaks_per_label` are different (the former is checked in the test above with the same test case). On the other hand, it doesn't check anything additional to `test_num_peaks_and_labels_4quadrants`.
I think you should merge `test_num_peaks_and_labels` and `test_num_peaks_per_labels`: for example, create image where the same values of `num_peaks` and `num_peaks_per_label` guarantee different results and put all 3 asserts into single test.